### PR TITLE
fix: add docs tag validation to solve #5478

### PR DESCRIPTION
**Pull request by sw-yx**
fix: add docs tag validation to solve #5478

## Motivation

nicer developer experience when people put in malformed tags, as suggested by @Josh-Cena  https://github.com/facebook/docusaurus/issues/5478#issuecomment-912118520

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

i confess i did not test this one locally. hoping its a simple one.

### CODE REVIEW

File: packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ const DocFrontMatterSchema = Joi.object<DocFrontMatter>({
   slug: Joi.string(),
   sidebar_label: Joi.string(),
   sidebar_position: Joi.number(),
+  tags: [Joi.string(), Joi.array().items(Joi.object({
+    label: Joi.string,
+    permalink: Joi.string
+  }))]
```
Comment (Josh-Cena) [inline packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts:33]:
```suggestion
  tags: Joi.array().items(Joi.alternatives().try(
    Joi.string(),
    Joi.object({
      label: Joi.string,
      permalink: Joi.string
    }),
  )),
```

Joi is really a pain, yeah

**Comment by Josh-Cena** [comment]
Also, maybe we can elaborate in the documentation? After some pondering I think it's worth documenting both possibilities to avoid confusion.

File: packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts (hunk 1 above)
Comment (Josh-Cena) [reply packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts:33]: Actually, there's a `FrontMatterTagsSchema` from `utils-validation`. See here: 

https://github.com/facebook/docusaurus/blob/fd3c0c661944e2aa88330f691e248b436b93b7f7/packages/docusaurus-plugin-content-blog/src/blogFrontMatter.ts#L8-L13

https://github.com/facebook/docusaurus/blob/fd3c0c661944e2aa88330f691e248b436b93b7f7/packages/docusaurus-plugin-content-blog/src/blogFrontMatter.ts#L78

Comment (sw-yx) [reply packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts:33]: thank you!

**Comment by sw-yx** [comment]
you mean this documentation? https://docusaurus.io/docs/create-doc#doc-tags 

i'm not sure its worth documenting both possibilities, if i read you right there is only one possibility?

**Comment by slorber** [comment]
Thanks,

I'll edit it a bit before merge because Joi default messages are so bad in this case :)

**Pull request merged**